.github/workflows/test.yml

```diff
@@ -71,3 +71,99 @@ jobs:
             Meadowlark-js/integration-tests/junit.xml
             Meadowlark-js/system-tests/junit.xml
           retention-days: 1
+
+  end-to-end:
+    name: End to End tests for ${{ matrix.config.db }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          [
+            { db: "Mongo", plugin: "@edfi/meadowlark-mongodb-backend" },
+            { db: "PGSQL", plugin: "@edfi/meadowlark-postgresql-backend" },
+          ]
+
+    defaults:
+      run:
+        working-directory: Meadowlark-js
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
+
+      - name: NPM caching
+        uses: actions/setup-node@17f8bd926464a1afa4c6a11669539e9c1ba77048 # v3.2.0
+        with:
+          node-version: "16"
+          cache: "yarn"
+          cache-dependency-path: "**/yarn.lock"
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Start OpenSearch Container
+        run: docker compose -f ./backends/meadowlark-opensearch-backend/docker/docker-compose.yml up -d
+
+      - if: ${{ matrix.config.db == 'PGSQL' }}
+        name: Start Postgres Container
+        run: docker compose -f ./backends/meadowlark-postgresql-backend/docker/docker-compose.yml up -d
+
+      - if: ${{ matrix.config.db == 'Mongo' }}
+        name: Start MongoDB
+        uses: supercharge/mongodb-github-action@538a4d2a1041920c47630172445cb688592d6e51 #v1.8.0
+        with:
+          mongodb-version: "4.0.28"
+          mongodb-replica-set: rs0
+          mongodb-port: 42069
+
+      - name: Cache Build
+        uses: actions/cache@48af2dc4a9e8278b89d7fa154b955c30c6aaab09 #v3.0.2
+        env:
+          cache-name: cache-build
+        with:
+          path: "**/dist/**"
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/dist/**') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+
+      - if: ${{ steps.cache-build.outputs.cache-hit != 'true' }}
+        name: Build
+        run: yarn build
+
+      - name: Run Fastify in background
+        run: |
+          cp ./tests/e2e/setup/e2e.env ./services/meadowlark-fastify/.env
+          yarn start:local > fastify.log &
```

Comment:
I like this log redirection, so that the GitHub Actions output isn't overwhelmed, but we still have access to the detail if needed.